src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8467,65 +8467,59 @@ private VariableComponentSyntax ParseDeconstructionComponent(bool topLevel = fal
             }
             else
             {
-                var type = ParseType(false);
+                TypeSyntax type;
+                bool reportMissingType = false;
+                if (this.CurrentToken.Kind == SyntaxKind.IdentifierToken &&
+                    (this.PeekToken(1).Kind == SyntaxKind.CommaToken || this.PeekToken(1).Kind == SyntaxKind.CloseParenToken))
```

Comment:
`DeconstructionTests.DeconstructionAssignment()` in the parsing tests exercises this.

---
In reply to: [75773794](https://github.com/dotnet/roslyn/pull/13087#discussion_r75773794) [](ancestors = 75773794)